File: JAVA/Two Repeated Elements.java
class Solution
{
    public int[] twoRepeated(int arr[], int n)
    {
       int[] res=new int[2];
       Set<Integer> s= new HashSet<>();
       for(int j=0;j<n+2;j++){
           if(s.contains(arr[j])){
              if(res[0]>0){
                   res[1]=arr[j];
               }
               else{
                   res[0]=arr[j];
               }
           }
           s.add(arr[j]);
       }
      return res;
    }
}

File: C++/Two Repeated Elements.cpp
class Solution {
  public:
    vector<int> twoRepeated (int arr[], int n) {
        vector<pair<int, int>>v;
        map<int, int>mp;
        for(int i=0;i<n+2;i++){
            mp[arr[i]]++;
            if( mp[arr[i]] > 1){
                v.push_back(make_pair(arr[i], i));
            }
        }
        vector<int>vec;
        if( v[1].second < v[0].second){
            vec.push_back(v[1].first);
            vec.push_back(v[0].first);
        }else{
            vec.push_back(v[0].first);
            vec.push_back(v[1].first);
        }
        return vec;
    }
};
